.github/workflows/file-ultils.yml

```
@@ -0,0 +1,56 @@
+# SPDX-License-Identifier: Apache-2.0
+# Licensed to the Ed-Fi Alliance under one or more agreements.
+# The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+# See the LICENSE and NOTICES files in the project root for more information.
+
+name: File Utilities Library
+on:
+  pull_request:
+    paths:
+      - 'src/file-utils/**'
+      - '!src/file-utils/**/*.md'
+  push:
+    paths:
+      - 'src/file-utils/**'
+      - '!src/file-utils/**/*.md'
```

@lalrg one problem we have is that we don't actually want to trigger on the push to the _feature_ branch - only on the push into the _main_ branch, if someone bypassed the pull request. So what we really want is to combine paths and branches here. Unfortunately I didn't get it working before. However, [this discussion post](https://github.community/t/combine-path-and-tags-conditionals/17064) claims that it is possible. So I am going to try it out here before merging this pull request.
```suggestion
      - '!src/file-utils/**/*.md'
    branches:
      - main
```